packages/astro/src/core/build/static-build.ts

```diff
@@ -377,6 +369,20 @@ async function generatePath(pathname: string, opts: StaticBuildOptions, gopts: G
 	const links = createLinkStylesheetElementSet(linkIds.reverse(), site);
 	const scripts = createModuleScriptElementWithSrcSet(hoistedId ? [hoistedId] : [], site);
 
+	// Add all injected scripts to the page.
+	for (const script of astroConfig._ctx.scripts) {
+		if (script.stage === 'head-inline') {
+			scripts.add({
+				props: {},
+				children: script.content,
+			});
+		}
+	}
+
+	// TODO: Add support for `injectElement()` for full HTML element injection, not just scripts.
```

Comment:
Shouldn't something like that be part of rendering and not part of the build? Otherwise it won't happen in SSR.

Reply:
I'm not sure I know what you mean, any links to a better location in the codebase for this TODO would be appreciated!

Reply:
Agreed! This is part of my concern with a single hook for "build" without a "compile" vs "render" discrepancy. I guess we're assuming that "build" implies static build.

Reply:
I'll move this TODO somewhere else that doesn't imply it will be handled by static build. And yes, I'm +1 on that. Would love to chat afterwards about the best path forward to implement with SSR in mind.